ERCS/erc-7589.md

```diff
@@ -0,0 +1,480 @@
+keep the standard agnostic of any implementation. This approach enables the standard to be implemented externally or on
+the same contract as the SFT and allows dApps to use roles with immutable SFTs.
+
+### Role Approvals
+
+Like [ERC-1155](./eip-1155.md), [ERC-7589](./eip-7589.md) allows users to approve operators to grant and revoke roles on
+their behalf. This feature is crucial for interoperability, as it enables third-party applications to manage user roles
+without custody-level approvals. Role approvals are part of the core interface, and compliant contracts MUST implement
+the `setRoleApprovalForAll` and `isRoleApprovedForAll` functions.
+
+### Automatic Expiration
+
+Automatic expiration is implemented to save users gas. To end a role assignment, instead of requiring users always to
+call `revokeRole`, applications SHOULD call the `roleExpirationDate` and compare it to the current timestamp to check if
+the role is still valid.
+
+In the context of [ERC-7589](./eip-7589.md), dates are represented as `uint64`. The maximum UNIX timestamp represented
+by a `uint64` is about the year 584 billion, which should be enough to be considered "permanent". For this reason, it's
+RECOMMENDED using `type(uint64).max` in use cases where the assignment is required never to expire.
+
+### Revocable Roles
+
+In certain scenarios, the grantor might need to revoke a role before its expiration. While in others, the grantee
+requires assurance that the role can't be prematurely revoked (e.g. when the grantee pays tokens to utilize them). The
+`_revocable` parameter was included in the `grantRole` function for this exact reason, and it specifies whether the
+grantor can revoke the role prior to the expiration date. Regardless of the `_revocable` value, the grantee SHOULD
+always be able to revoke roles, allowing recipients to eliminate undesirable assignments.
+
+### Custom Data
+
+The `grantRole` function's `_data` parameter is critical for the standardization of this EIP. SFTs have different use
+cases, and it's impractical to attempt to cover all of them on a solidity-level interface. Therefore, a generic
+parameter of type `bytes` was incorporated, allowing users to pass any custom information when granting a role.
+
+For example, it's common for web3 games to introduce a profit-share when delegating NFTs to players, which is
+represented by a `uint256`. Using [ERC-7589](./eip-7589.md), one could simply encode the `uint256` as bytes and pass it
+to the`grantRole` function. Data validation can happen on-chain or off-chain, and other contracts can query this
+information using the `roleData` function.
+
+
+## Backwards Compatibility
+
+Many SFTs are deployed as immutable contracts, which imposes the following challenge: How can one enable role management
+for SFTs that can't be modified? [ERC-7589](./eip-7589.md) solves this problem by requiring the `tokenAddress` parameter
```

Comment:
Bit of a stylistic recommendation, but instead of linking to itself, just say "this proposal".